src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.AgentPool.ps1

```diff
@@ -425,6 +479,15 @@ class AgentPool: ADOSVTBase
                         $display=($agentsWithSecretsInEnv | FT AgentName,Capabilities -AutoSize | Out-String -Width 512)
                         $controlResult.AddMessage($display)
                         $controlResult.SetStateData("Agent-wise list of user-defined capabilities with secrets: ", $agentsWithSecretsInEnv );
+                        $backupDataObject= @()
+                        @($agentDetails.Keys) | ForEach-Object {
+                            $key = $_
+                            $obj = '' | Select @{l="PoolId";e={$poolId}}, @{l="AgentId";e={$key}},@{l="CapabilitiesWithSecret";e={($agentDetails.item($key))[0]}}, @{l="CapabilitiesWithoutSecret";e={($agentDetails.item($key))[1]}}
```

Comment:
We should not store the secret value in plain text on the local machine. Please encrypt it here, and decrypt it while using it in auto fix method. Also there seems to be some flaw in backupobj. Capabilities without secrets are also being added in withsecrets var:
"CapabilitiesWithoutSecret" : {
                    "test2" : "abc"
              }, 
              "CapabilitiesWithSecret" : {
                  "secret" : "Password123!@#", 
                  "test2" : "abc"
              }